qualtran/_infra/composite_bloq.py

```diff
@@ -1042,6 +1046,18 @@ def add_and_partition(
 
         return self.add(AutoPartition(bloq, partitions, left_only), **in_soqs)
 
+    def add_d_and_partition(
+        self,
+        bloq: Bloq,
+        partitions: Sequence[Tuple[Register, Sequence[Union[str, 'Unused']]]],
+        left_only: bool = False,
+        **in_soqs: SoquetInT,
+    ) -> Dict[str, SoquetT]:
+        """Version of `add_and_partition` that returns a dictionary. See `BloqBuilder.add_d`."""
+        from qualtran.bloqs.bookkeeping.auto_partition import AutoPartition
+
+        return self.add_d(AutoPartition(bloq, partitions, left_only), **in_soqs)
```

Comment:
I'm a little worried about continually expanding the API surface of `BloqBuilder` by having all these variants as methods, especially when it's just straightforwardly passing through each argument to existing methods. In this case, I think the one-liner `self.add_d(AutoPartition(...), ...)` is straightforward enough to use when writing decompositions directly. What do you think?

Reply:
Yes, that is fair. I backed this out.